src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts

```diff
@@ -922,23 +921,17 @@ export class SimpleFileDialog {
 		return true;
 	}
 
-	private async createItem(filename: string, parent: URI, token: CancellationToken): Promise<FileQuickPickItem | undefined> {
+	private async createItem(stat: IFileStat, parent: URI, token: CancellationToken): Promise<FileQuickPickItem | undefined> {
 		if (token.isCancellationRequested) {
 			return undefined;
 		}
-		let fullPath = resources.joinPath(parent, filename);
-		try {
-			const stat = await this.fileService.resolve(fullPath);
-			if (stat.isDirectory) {
-				filename = resources.basename(fullPath);
-				fullPath = resources.addTrailingPathSeparator(fullPath, this.separator);
-				return { label: filename, uri: fullPath, isFolder: true, iconClasses: getIconClasses(this.modelService, this.modeService, fullPath || undefined, FileKind.FOLDER) };
-			} else if (!stat.isDirectory && this.allowFileSelection && this.filterFile(fullPath)) {
-				return { label: filename, uri: fullPath, isFolder: false, iconClasses: getIconClasses(this.modelService, this.modeService, fullPath || undefined) };
-			}
-			return undefined;
-		} catch (e) {
-			return undefined;
+		let fullPath = resources.joinPath(parent, stat.name);
+		if (stat.isDirectory) {
+			fullPath = resources.addTrailingPathSeparator(fullPath, this.separator);
+			return { label: resources.basename(fullPath), uri: fullPath, isFolder: true, iconClasses: getIconClasses(this.modelService, this.modeService, fullPath || undefined, FileKind.FOLDER) };
```

Comment:
Still should do `resuorces.basename(fullPath)` before the `addTrailingPathSeparator` since you can end up with an extra path separator.